src/annotator/annotation-sync.js

```diff
@@ -1,176 +1,116 @@
-// AnnotationSync listens for messages from the sidebar app indicating that
-// annotations have been added or removed and relays them to Guest.
-//
-// It also listens for events from Guest when new annotations are created or
-// annotations successfully anchor and relays these to the sidebar app.
-export default function AnnotationSync(bridge, options) {
-  const self = this;
-
-  this.bridge = bridge;
-
-  if (!options.on) {
-    throw new Error('options.on unspecified for AnnotationSync.');
-  }
-
-  if (!options.emit) {
-    throw new Error('options.emit unspecified for AnnotationSync.');
-  }
-
-  this.cache = {};
-
-  this._on = options.on;
-  this._emit = options.emit;
-
-  // Listen locally for interesting events
-  Object.keys(this._eventListeners).forEach(function (eventName) {
-    const listener = self._eventListeners[eventName];
-    self._on(eventName, function (annotation) {
-      listener.apply(self, [annotation]);
+/**
+ * @typedef {import('../shared/bridge').default} Bridge
+ */
+
+/**
+ * @callback RpcCallback
+ * @param {Error|null} error
+ * @param {any} result
+ */
+
+/**
+ * AnnotationSync listens for messages from the sidebar app indicating that
+ * annotations have been added or removed and relays them to Guest.
+ *
+ * It also listens for events from Guest when new annotations are created or
+ * annotations successfully anchor and relays these to the sidebar app.
+ */
+export default class AnnotationSync {
+  /**
+   * @param {Bridge} bridge
+   * @param {Object} options
+   * @param {(event: string, callback: (data: any, callback: RpcCallback) => void) => void} options.on -
```

Comment:
Oof, typedefs can get tricksy to human-parse sometimes :).  Should this and the following property of `options` be indented?

Reply:
Reviewing this again this morning it turns out that we don't actually use the `callback: RpcCallback` argument so I'll simplify things by removing it.

To your question, yes this should have been indented.